docs/docs/domain.mdx

```diff
@@ -616,20 +732,17 @@ call, or to start the conversation with a bot message. The docs on
 ## Config
 
 The `config` key in the domain file maintains the `store_entities_as_slots` parameter.
-When an entity is recognized by the NLU model and the entity name matches a slot name,
-`store_entities_as_slots` defines whether the entity value should be placed in that slot.
-By default, entities will auto-fill slots of the same name.
+When an entity is recognized by the NLU model and it matches the `from_entity` slot mapping, `store_entities_as_slots`
+defines whether the entity value should be placed in that slot.
+By default, this behaviour is switched on.
 
-You can turn off all slot auto-filling by setting the `store_entities_as_slots` parameter to `false`:
+You can turn off this functionality by setting the `store_entities_as_slots` parameter to `false`:
```

Comment:
with `auto_fill` the turning off made sense, but in the new scenario, what's the point of controlling this here? Doesn't it implicitly nullify the `from_entity` mappings, so one could just as well delete them?
I'm sure there's a good reason, just think we should tell users what it is

Reply:
Yes, you're right, this doesn't make sense that much anymore; i only kept it because this attribute (`store_entities_as_slots`) conditions the running of a Domain method (`slots_for_entities()`) that gets called in `StoryStep.explicit_events`. So it helps with a story such as the one debated in a thread a while ago:
```
- story: example story
  steps:
  - or:
     - intent: greet
     - intent: greet
        entities:
        - name: Molly
 - action: utter_greet
```
Probably an idea to mention that if this parameter gets turned off the stories won't work accordingly 🤔

Reply:
Ah right, I remember that thread now. 

Not sure this is clarified yet though: Does `store_entities_as_slots: false` == removing all `from_entity` mappings?
Or, is there some way the `from_entity` mappings get applied, even when `store_entities_as_slots: false`?

 Does this parameter control how the _syntax_ in stories is interpreted at training time, and is this any different from how user messages are interpreted & used to set slots at inference time?
e.g.
if `store_entities_as_slots: true`, then this story and the following one are equivalent:
```
- story: example story
  steps:
  - or:
     - intent: greet
     - intent: greet
        entities:
        - name: Molly
 - action: utter_greet
```
==
```
- story: example story
  steps:
  - or:
     - intent: greet
     - intent: greet
        entities:
        - name: Molly
     - slot_was_set:
        - name: Molly
 - action: utter_greet
```

If `store_entities_as_slots: false`, then the first story is rather equivalent to this other one:

```
- story: example story
  steps:
  - or:
     - intent: greet
     - intent: greet
        entities:
        - name: Molly
     - slot_was_set:
        - name: null
 - action: utter_greet
```

Reply:
An example of the impact of this parameter is definitely needed here, I've ran into A Lot of user confusion on this behaviour.

Reply:
> Does store_entities_as_slots: false == removing all from_entity mappings?
> Or, is there some way the from_entity mappings get applied, even when store_entities_as_slots: false?

No, this will not affect how `from_entity` slots are filled during the conversation (meaning the default action will carry on running just fine).

> Does this parameter control how the syntax in stories is interpreted at training time, and is this any different from how user messages are interpreted & used to set slots at inference time?

Yes, this parameter will only control to how stories are interpreted at training time. It doesn't influence how slots are set at inference time after every user turn.

I don't know for sure if that exact story syntax you listed for `store_entities_as_slots: false` works, but  it's in the right direction: if this parameter gets turned off, then the tracker won't have a `SlotSet` event for the entity in the user message listed in the story.

Reply:
Ok! Then that's what I think we should explain in this section,  that this parameter is only relevant to reading stories and turning them into trackers, and making slot set events out or entities.